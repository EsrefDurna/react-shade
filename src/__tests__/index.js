import React from 'react';
import { render as reactDomRender } from 'react-dom';
import { renderIntoDocument } from 'react-dom/test-utils';
import ShadowRoot from '..';

document.body.innerHTML = '<div></div>';

function render(node) {
  const { firstChild } = document.body;
  reactDomRender(node, firstChild);

  // The wrapper creates a wrapping node because ReactDOM's createPortal()
  // function can't render to a shadow root, so we must reach in and return
  // the wrapper it generates.
  return firstChild.firstChild;
}

test('creates a shadow root', () => {
  const root = render(<ShadowRoot />);
  expect(root.shadowRoot).toBeDefined();
});

test('renders text into the shadow root', () => {
  const root = render(<ShadowRoot>test</ShadowRoot>);
  expect(root.shadowRoot.innerHTML).toBe('<div>test</div>');
});

test('renders elements into the shadow root', () => {
  const root = render(
    <ShadowRoot>
      <div>test</div>
    </ShadowRoot>
  );
  expect(root.shadowRoot.innerHTML).toBe('<div><div>test</div></div>');
});

test('rendered inside of another node', () => {
  const root = render(
    <section>
      <ShadowRoot>
        <div>test</div>
      </ShadowRoot>
    </section>
  );
  expect(root.nodeName).toBe('SECTION');
  expect(root.firstChild.shadowRoot.innerHTML).toBe(
    '<div><div>test</div></div>'
  );
});

test('props - tag', () => {
  const root = render(<ShadowRoot tag="span">test</ShadowRoot>);
  expect(root.nodeName).toBe('SPAN');
  expect(root.shadowRoot.innerHTML).toBe('<span>test</span>');
});
