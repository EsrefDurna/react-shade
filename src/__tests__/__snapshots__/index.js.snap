// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slot - renders a slot 1`] = `"<div></div>"`;

exports[`Slot - renders a slot 2`] = `"<shadow--root><slot name=\\"slot-1\\"></slot></shadow--root>"`;

exports[`Slot - renders multiple slots 1`] = `"<div></div>"`;

exports[`Slot - renders multiple slots 2`] = `"<shadow--root><slot name=\\"slot-1\\"></slot><slot name=\\"slot-2\\"></slot><slot name=\\"slot-3\\"></slot></shadow--root>"`;

exports[`Slot - slots content 1`] = `"<div><span slot=\\"slot-1\\">content 1</span><span slot=\\"slot-2\\">content 2</span><span slot=\\"slot-3\\">content 3</span></div>"`;

exports[`Slot - slots content 2`] = `"<shadow--root><slot name=\\"slot-1\\"></slot><slot name=\\"slot-2\\"></slot><slot name=\\"slot-3\\"></slot></shadow--root>"`;

exports[`Slot - slots content 3`] = `"<div><div><span slot=\\"slot-1\\">content 1</span><span slot=\\"slot-2\\">content 2</span><span slot=\\"slot-3\\">content 3</span></div></div>"`;

exports[`props - tag 1`] = `"<span></span>"`;

exports[`props - tag 2`] = `"<shadow--root>test</shadow--root>"`;

exports[`rendered inside of another node 1`] = `"<section><div></div></section>"`;

exports[`rendered inside of another node 2`] = `"<shadow--root><div>test</div></shadow--root>"`;

exports[`renders elements into the shadow root 1`] = `"<shadow--root><div>test</div></shadow--root>"`;

exports[`renders text into the shadow root 1`] = `"<shadow--root>test</shadow--root>"`;
